soccer/gameplay/plays/restarts/our_shootout_chip.py

```diff
@@ -0,0 +1,182 @@
+                            self.subbehavior_with_name('chipping').state == behavior.Behavior.State.completed,
+                            'recapture')
+
+        # after you capture shoot the ball with a kick
+        self.add_transition(OurShootoutChip.State.capture,
+                            OurShootoutChip.State.shooting,
+                            lambda:self.subbehavior_with_name('capture').state == behavior.Behavior.State.completed,
+                            'reshoot')
+
+        self.total_time = 10.0
+        self.shoot_time = 6.0
+        self.start_time = time.time()
+        self.chip_distance = 1.5
+        self.goalie_range = 2
+
+    @classmethod
+    def handles_goalie(cls):
+        return True
+
+    @classmethod
+    def is_restart(cls):
+        return False
+
+    @classmethod
+    def score(cls):
+        gs = main.game_state()
+        randomfloat = random.random()
+        return randomfloat if gs.is_penalty_shootout() and gs.is_our_penalty() else float("inf")
+
+    def on_enter_starting(self):
+        # find the direction from the enemy goal to the ball
+        behind_ball = (main.ball().pos - constants.Field.TheirGoalSegment.center())
+        # normalize the vector
+        behind_ball = behind_ball.normalized()
+        # move the robot one robot width behind the ball away from the opponent's goal
+        start_point = behind_ball * 2 * constants.Robot.Radius + main.ball().pos
+        self.add_subbehavior(skills.move.Move(start_point), 'starting', required = False,
+                            priority = 5)
+
+    def on_exit_starting(self):
+        self.remove_all_subbehaviors()
+
+    def on_enter_dribbling(self):
+        # set the time for the start of the play
+        self.start_time = time.time()
+        # find the vector of the ball to the goal
+        robot_pos = main.ball().pos
+        center_goal = constants.Field.TheirGoalSegment.center()
+        dribble_point = (center_goal - robot_pos)
+        # normalize the vector
+        dribble_point = dribble_point.normalized()
+        # added that to the current ball position going one meter to the goal
+        dribble_point = dribble_point + robot_pos
+        # dribble one meter up
+        dribble = skills.dribble.Dribble(dribble_point)
+        if (not self.has_subbehavior_with_name('dribble')):
+            self.add_subbehavior(dribble, 'dribble', required = False, priority = 5)
+
+    def on_exit_dribbling(self):
+        self.remove_all_subbehaviors()
+
+    def on_enter_chipping(self):
+            # setup chipper
+            kicker = skills.line_kick.LineKick()
+            kicker.use_chipper = True
+            kicker.min_chip_range = our_free_kick.OurFreeKick.MinChipRange
+            kicker.max_chip_range = our_free_kick.OurFreeKick.MaxChipRange
+            #calculate target
+            gap = evaluation.shooting.find_gap(max_shooting_angle= our_free_kick.OurFreeKick.MaxShootingAngle)
```

Comment:
Find gap is more for free kicks, the technically correct way to do this is win eval. Find gap does use win eval, but there's some extra stuff that we don't want here